### PR TITLE
Fix supported provider_settings in docs

### PR DESCRIPTION
**Pull request by vgrigoruk**
This was discovered during implementation of https://github.com/buildkite/terraform-provider-buildkite/pull/123

### CODE REVIEW

**Comment by plaindocs** [comment]
Hey @vgrigoruk thanks for the pull request. 

Can I just check that deleting `filter_enabled` was intentional, as well as changing the values of `publish_blocked_as_pending`? Cheers.

**Comment by vgrigoruk** [comment]
Hi @plaindocs . Yes, the change is based on t fact that API doesn't support the deleted properties (they look like a copy-paste from other doc). Also, publish_blocked_as_pending is boolean.

**Comment by plaindocs** [comment]
I did a little bit of digging, and it seems that `filter_enabled` is behind a feature flag, and I'm seeing if I can either get it released, or document it as such. I'll keep you posted.

File: pages/apis/rest_api/pipelines.md.erb [hunk 1]
```diff
@@ -687,19 +687,6 @@ Additional properties available for GitHub:
         What type of event to trigger builds on. <code>Code</code> will create builds when code is pushed to GitHub. <code>Deployment</code> will create builds when a deployment is created with the <a href="https://developer.github.com/v3/repos/deployments/">GitHub Deployments API</a>. <code>Fork</code> will create builds when the GitHub repository is forked. <code>None</code> will not create any builds based on GitHub activity.
         <p class="Docs__api-param-eg"><em>Values:</em> <code>code</code>, <code>deployment</code>, <code>fork</code>, <code>none</code></p></td>
     </tr>
```
Comment (plaindocs) [inline pages/apis/rest_api/pipelines.md.erb:689]:
```suggestion
    </tr>
    <tr>
      <th><code>filter_enabled</code></th>
      <td>
        Whether filter conditions are being used for this step.
        <p class="Docs__api-param-eg"><em>Values:</em> <code>true</code>, <code>false</code></p></td>
    </tr>
    <tr>
      <th><code>filter_condition</code></th>
      <td>
        The conditions under which this step will run. See the <a href="/docs/pipelines/conditionals">Using Conditionals</a> guide for more information.
        <p class="Docs__api-param-eg"><em>Example:</em> <code>"build.message != 'skip'"</code></p>
      </td>
    </tr>        
```

Comment (plaindocs) [reply pages/apis/rest_api/pipelines.md.erb:689]: @vgrigoruk let's put these back, I'll open a separate PR when I hear the situation resolved, and we can merge your fixes to `publish_blocked_as_pending`

**Review by plaindocs — APPROVED** [review]

**Pull request merged**